### PR TITLE
Add runfiles library to release archive

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by hvadehra** [comment]
I'm going to making another release shortly, so happy to merge this now, and get the test in later.

**Review by hvadehra — APPROVED** [review]

**Review by hvadehra — CHANGES_REQUESTED** [review]
Please remove the "for_bazel_tests" changes

**Comment by fmeum** [comment]
> How urgent is this?

It's needed for https://github.com/bazelbuild/bazel/pull/24219. It's not a hard requirement for Bazel 8 from my point of view, but it could speed up the removal of ruleset deps from `bazel_tools` if it does get into the release.

> Do we want to add an integration test in https://github.com/bazelbuild/rules_java/tree/master/test/repo that would have caught something like this?

I wasn't aware of this setup already existing. I will add a test as part of this PR as it is easy enough.

**Comment by fmeum** [comment]
@hvadehra I added a test.

**Comment by fmeum** [comment]
@hvadehra I ran `google-java-format` on the file, hope that solves the issues.

File: test/repo/src/MyTest.java [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 @RunWith(JUnit4.class)
+@AutoBazelRepository
 public class MyTest {
   @Test
   public void main() {
     assertEquals(MyLib.myStr(), "my_string");
   }
-}
 
+  @Transient
```
[the remark this answers is not in the document]
Comment (fmeum) [reply test/repo/src/MyTest.java]: Yes, fixed

**Pull request closed without merging**